File: bindings/nodejs/src/clients/versionRpcClient.js
const { RpcSession } = require("../rpcSession");
const { RpcClientError } = require("../errors");

class CenyslovenskoVersionRpcClient {
  constructor(options = {}) {
    this.session = options.session || new RpcSession(options);
  }

  start() {
    this.session.start();
  }

  close() {
    this.session.close();
  }

  async call(method, params) {
    return this.session.call(method, params);
  }

  async getVersion() {
    const response = await this.call("version.get");
    if (typeof response.version !== "string") {
      throw new RpcClientError("Missing or invalid 'version' in response");
    }
    return response.version;
  }
}

module.exports = {
  CenyslovenskoVersionRpcClient,
};

File: bindings/ruby/lib/cenyslovensko_bindings/clients/version_rpc_client.rb
module CenyslovenskoBindings
  module Clients
    class VersionRpcClient
      def initialize(session: nil, **session_options)
        @session = session || RpcSession.new(**session_options)
      end

      def start
        @session.start
      end

      def close
        @session.close
      end

      def call(method, params = nil)
        @session.call(method, params)
      end

      def get_version
        response = call("version.get")
        version = response["version"]
        raise RpcClientError, "Missing or invalid 'version' in response" unless version.is_a?(String)

        version
      end
    end
  end
end
